lib/smf/smf.c

```diff
@@ -9,24 +9,25 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(smf);
 
-/*
- * Private structure (to this file) used to track state machine context.
- * The structure is not used directly, but instead to cast the "internal"
- * member of the smf_ctx structure.
+/**
+ * @brief Private structure (to this file) used to track state machine context.
+ *        The structure is not used directly, but instead to cast the "internal"
+ *        member of the smf_ctx structure.
  */
 struct internal_ctx {
 	bool new_state : 1;
 	bool terminate : 1;
 	bool exit      : 1;
 	bool handled   : 1;
+#ifdef CONFIG_SMF_UML_HSM
+	bool entry     : 1;
+#endif
 };
 
-static bool share_paren(const struct smf_state *test_state,
-			const struct smf_state *target_state)
+__unused static bool share_paren(const struct smf_state *test_state,
+				 const struct smf_state *target_state)
 {
-	for (const struct smf_state *state = test_state;
-	     state != NULL;
-	     state = state->parent) {
+	for (const struct smf_state *state = test_state; state != NULL; state = state->parent) {
```

Comment:
There's still a bunch of changes caused by me running clang-format on the file. Should I leave them?